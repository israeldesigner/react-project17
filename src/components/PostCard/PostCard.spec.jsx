import { render, screen } from '@testing-library/react';
import { PostCard } from './index';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render POstCard correct', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: 'title1' })).toBeInTheDocument();
    expect(screen.getByText('body1')).toBeInTheDocument();
  });

  it('should mathc snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
