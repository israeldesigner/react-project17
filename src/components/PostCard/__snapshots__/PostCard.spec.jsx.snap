// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should mathc snapshot 1`] = `
<div
  class="post"
>
  <div
    class="post-thumbnail"
  >
    <img
      alt="title1"
      class="post-img"
      src="img/img.png"
    />
  </div>
  <div
    class="post-content"
  >
    <h1>
      title1
    </h1>
    <p>
      body1
    </p>
  </div>
</div>
`;
